src/python/WMQuality/Emulators/RucioClient/MockRucioApi.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Version of WMCore/Services/Rucio intended to be used with mock or unittest.mock
+"""
+from __future__ import print_function, division
+
+from RestClient.ErrorHandling.RestClientExceptions import HTTPError
+
+from WMCore.Services.DBS.DBS3Reader import DBS3Reader, DBSReaderError
+from WMCore.Services.Rucio.Rucio import WMRucioException, WMRucioDIDNotFoundException
+from WMQuality.Emulators.DataBlockGenerator.DataBlockGenerator import DataBlockGenerator
+
+PROD_DBS = 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader'
+
+NOT_EXIST_DATASET = 'thisdoesntexist'
+PILEUP_DATASET = '/HighPileUp/Run2011A-v1/RAW'
+
+SITES = ['T2_XX_SiteA', 'T2_XX_SiteB', 'T2_XX_SiteC']
+_BLOCK_LOCATIONS = {}
+
+BLOCKS_PER_DATASET = 2
+FILES_PER_BLOCK = 5
+FILES_PER_DATASET = BLOCKS_PER_DATASET * FILES_PER_BLOCK
+
+MOCK_DATA = {}
+
+
+class MockRucioApi(object):
+    """
+    Version of WMCore/Services/Rucio intended to be used with mock or unittest.mock
+    """
+
+    def __init__(self, acct, hostUrl=None, authUrl=None, configDict=None):
+        print("Using MockRucioApi: acct={}, url={}, authUrl={}".format(acct, hostUrl, authUrl))
+        configDict = configDict or {}
+        self.dataBlocks = DataBlockGenerator()
+        self.subRequests = {}
+
+    def sitesByBlock(self, block):
+        """
+        Centralize the algorithm to decide where a block is based on the hash name
```

Comment:
Alan, sorry it may be just me but still I think it wouldn't hurt if this doc string is made a little bit more clear.

Reply:
Can certainly be done! It was basically a copy of the PhEDExMock class.

Reply:
Thanks, anyway. We may do it at anytime - no need it to happen in this very moment.